hr_holidays_hour/models/hr_employee.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2017-2019 Onestein (<https://www.onestein.eu>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class HrEmployee(models.Model):
+    _inherit = "hr.employee"
+
+    @api.multi
+    def _compute_leaves_count(self):
+        leaves = self.env['hr.leave'].read_group([
```

Comment:
@astirpe  here use `search_count` method, So easy  counting and fast return result

same as for allocations

Reply:
And even better to use `read_group` for batch record compute.

Reply:
@nikul-serpentcs I don't understand why should I use `search_count`. In method `_compute_leaves_count()` I'm not counting the number of records, I'm counting the hours of leaves and allocations.

@pedrobaeza I also don't understand what are you proposing.

Reply:
> @nikul-serpentcs I don't understand why should I use `search_count`. In method `_compute_leaves_count()` I'm not counting the number of records, I'm counting the hours of leaves and allocations.


Ok, Or You can search a record which is your Domain and sum of `number_of_hours`.
It's Not Blocking anywhere but improvements.